### ISSUE TITLE
How to use pretrained RL model to generate RGBD data?

### ISSUE THREAD

**Issue by Cwb0106** [arth-shukla/mshab#44]
Hello, I have another qusetion.

I want to use your RL checkpoint to generate RGBD data. But when I simply modify FetchDepthObservationWrapper->FetchRGBDObservationWrapper (return xxx, **depth_pixels, **rgb_pixels) , it still only keeps the depth data without RGB

`class FetchRGBDObservationWrapper(gym.ObservationWrapper):
    def __init__(self, env, cat_state=True, cat_pixels=False) -> None:
        super().__init__(env)

        self.cat_pixels = cat_pixels
        self.cat_state = cat_state
        self._stack_fn = torch.stack
        self._cat_fn = torch.cat

        self._base_env: BaseEnv = env.unwrapped
        init_raw_obs = common.to_tensor(self._base_env._init_raw_obs)

        self._base_env.update_obs_space(common.to_numpy(self.observation(init_raw_obs)))

    def observation(self, observation):
        agent_obs = observation["agent"]
        extra_obs = observation["extra"]
        fetch_head_depth = observation["sensor_data"]["fetch_head"]["depth"].permute(
            0, 3, 1, 2
        )
        fetch_hand_depth = observation["sensor_data"]["fetch_hand"]["depth"].permute(
            0, 3, 1, 2
        )

        fetch_head_rgb = observation["sensor_data"]["fetch_head"]["rgb"].permute(
            0, 3, 1, 2
        )
        fetch_hand_rgb = observation["sensor_data"]["fetch_hand"]["rgb"].permute(
            0, 3, 1, 2
        )

        rgbd_pixels = (
            dict(
                all_depth=self._stack_fn([fetch_head_depth, fetch_hand_depth], axis=-3),
                all_rgb=self._stack_fn([fetch_head_rgb, fetch_hand_rgb], axis=-3)
            )
            if self.cat_pixels
            else dict(
                fetch_head_depth=fetch_head_depth,
                fetch_hand_depth=fetch_hand_depth,
                fetch_head_rgb=fetch_head_rgb,
                fetch_hand_rgb=fetch_hand_rgb,
            )
        )
        return (
            dict(
                state=self._cat_fn(
                    [
                        flatten_state_dict(agent_obs, use_torch=True),
                        flatten_state_dict(extra_obs, use_torch=True),
                    ],
                    axis=1,
                ),
                **rgbd_pixels,
            )
            if self.cat_state
            else dict(
                agent=agent_obs,
                extra=extra_obs,
                **rgbd_pixels,
            )
        )`

When I add “fetch_head_rgb” to the stacking_keys , SAC_agent throws an error. Is there something I'm missing? What should I modify so I can use the RL checkpoints you provided to collect RGBD data?
